### PR TITLE
[MODORDSTOR-379] - Manage updates of custom fields through RecordServiceImpl

### PR DESCRIPTION
**Pull request by alb3rtino**
## Purpose
https://folio-org.atlassian.net/browse/MODORDSTOR-379

This pull request switches the implementation used for `RecordService` from the default `NoOpRecordService` to `RecordServiceImpl`. See the [README](https://github.com/folio-org/folio-custom-fields?tab=readme-ov-file#recordservice) section in [folio-custom-fields](https://github.com/folio-org/folio-custom-fields).

`RecordServiceImpl` ensures that if a user deletes or updates a custom field, these changes will be appropriately reflected in the entities that have values set for this field. Basically, if a custom field is removed it also gets removed from its entities. If a selectable option is removed from a custom field then its entities will no longer have that option selected. 

## Approach
- Implemented `RecordServiceFactory` to return an instance of `RecordServiceImpl`.
- Added tests to validate the functionality of the implementation.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  @Override
+  public RecordService create(Vertx vertx) {
+    return RecordServiceImpl.createForMultipleTables(
+      vertx,
+      Map.of(
+        "po_line",
+        List.of("po_line", "order_templates"),
+        "purchase_order",
+        List.of("purchase_order", "order_templates")));
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: It looks like this implementation makes many calls in orders tables with this query: WHERE jsonb->'customFields' ? $1
In this case can you check if index was added for custom fields that will cover this query?
You can check example how to add indexes in schema.json.
So yo ujust need to update that file and RMB will handle index creating

Comment (alb3rtino) [reply src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: Good point. I'll check.

Comment (alb3rtino) [reply src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: Creating indexes using RMB seems unsuitable in this case.
Adding to `schema.json` for table `purchase_order`:
```
"ginIndex": [
        {
          "fieldName": "customFields",
          "caseSensitive": true,
          "removeAccents": false
        }
      ]
```
I was only able to create the following index: 
```
"purchase_order_customfields_idx_gin" gin ((jsonb ->> 'customFields'::text) gin_trgm_ops)
```

Which does not cover the query.

Instead I manually added a `gin ((jsonb -> 'customFields'::text))` index to every table effected by this PR - `purchase_order`, `po_line` and `order_templates`.

explain_analyze

Comment (alb3rtino) [reply src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: The index creation worked with RMB, but the created index does not cover queries that use the `?` JSONB operator.

Comment (SerhiiNosko) [reply src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: can you also please ask this question in #raml-module-builder slack channel, if this is not supported, we will go with approach to manually create indexes as implemented now in the PR

Comment (alb3rtino) [reply src/main/java/org/folio/services/record/RecordServiceFactoryImpl.java:22]: I can do that.

**Review by SerhiiNosko — APPROVED** [review]

**Review by damien-git — APPROVED** [review]

**Pull request merged**